packages/react/cypress/component/auto/form/PolarisAutoEnumInput.cy.tsx

```diff
@@ -0,0 +1,295 @@
+        allowOtherType: true,
+      });
+
+      cy.mountWithWrapper(<PolarisAutoForm action={api.game.stadium.create} />, PolarisWrapper);
+      cy.get("#type-combobox-textfield").type("hello");
+      cy.contains(`Add "hello"`).click();
+      cy.get("#type-combobox-textfield").should("have.value", "hello");
+
+      cy.get("#tags-combobox-textfield").type("world");
+      cy.contains(`Add "world"`).click();
+      cy.get(".Polaris-InlineStack").should("contain", "world");
+
+      blurComboboxes();
+
+      cy.getSubmitButton().click();
+      cy.get("@createStadium")
+        .its("request.body.variables.stadium")
+        .should("deep.equal", {
+          type: "hello",
+          tags: ["world"],
+        });
+    });
+
+    it("should not allow adding an option if allowOther is disabled", () => {
+      interceptModelActionMetadata({
+        allowOtherTags: false,
+        allowOtherType: false,
+      });
+
+      cy.mountWithWrapper(<PolarisAutoForm action={api.game.stadium.create} />, PolarisWrapper);
+      cy.get("#type-combobox-textfield").type("hello");
+      cy.contains(`Add "hello"`).should("not.exist");
+      cy.contains(`No found matching "hello"`).should("exist");
+
+      blurComboboxes();
+
+      cy.get("#tags-combobox-textfield").type("world");
+      cy.contains(`Add "world"`).should("not.exist");
+      cy.contains(`No found matching "world"`).should("exist");
+    });
+  });
+
+  describe("pre-filled value", () => {
+    beforeEach(() => {
+      cy.intercept(
+        {
+          method: "POST",
+          url: `${api.connection.endpoint}?operation=stadium`,
+        },
+        {
+          data: getStadiumRecord(),
+        }
+      ).as("getStadium");
+    });
+
+    it("should automatically set the pre-filled value when the form is pre-filled", () => {
+      interceptModelActionMetadata({ actionType: "update" });
+
+      cy.mountWithWrapper(<PolarisAutoForm action={api.game.stadium.update} findBy="42" />, PolarisWrapper);
+      cy.get("#type-combobox-textfield").should("have.value", "football");
+      cy.get(".Polaris-InlineStack").should("contain", "foo");
+      cy.get(".Polaris-InlineStack").should("contain", "bar");
+    });
+
+    it("should be able to clear the value", () => {
+      interceptModelActionMetadata({ actionType: "update" });
+      cy.mountWithWrapper(<PolarisAutoForm action={api.game.stadium.update} findBy="42" />, PolarisWrapper);
+      cy.get("#type-combobox-textfield").click();
+      cy.contains("football").click();
+
+      blurComboboxes();
+
+      cy.get("#tags-combobox-textfield").click();
+      cy.get('[aria-label="Remove foo"]').click();
+      cy.get('[aria-label="Remove bar"]').click();
+
+      blurComboboxes();
+
+      cy.getSubmitButton().click();
+      cy.get("@updateStadium").its("request.body.variables.stadium").should("deep.equal", {
+        type: null,
+        tags: [],
+      });
+    });
+  });
+});
+
+const interceptModelActionMetadata = (options?: {
```

Comment:
I think we should try to avoid helpers like this which take a lot of different options. I think this ends up being pretty complicated and hard to reason about what the test is actually trying to do. 

Another pattern is to have a set of mock objects which extend each other. For example:

```
StadiumBaseMock = {
  ...base stuff
}

StadiumCreateMock = {...StadiumBaseMock, {action: {apiIdentifier: 'create', operatesWithRecordIdentity: false}}}

```

And then your test can import the mock that it needs.